easy-ui-react/src/CodeBlock/CodeBlock.test.tsx

```diff
@@ -1,53 +1,98 @@
-import { render, screen } from "@testing-library/react";
+import { act, screen } from "@testing-library/react";
 import React from "react";
-import { mockGetComputedStyle } from "../utilities/test";
+import { vi } from "vitest";
+import { mockGetComputedStyle, render } from "../utilities/test";
 import { CodeBlock } from "./CodeBlock";
 
 describe("<CodeBlock />", () => {
   let restoreGetComputedStyle: () => void;
 
   beforeEach(() => {
+    vi.useFakeTimers();
     restoreGetComputedStyle = mockGetComputedStyle();
   });
 
   afterEach(() => {
     restoreGetComputedStyle();
+    vi.useRealTimers();
   });
 
   it("should render a code block", () => {
     render(
-      <CodeBlock code={`console.log("Hello world!");`} language="javascript" />,
+      <CodeBlock language="javascript" onLanguageChange={() => {}}>
+        <CodeBlock.Header>Header</CodeBlock.Header>
+        <CodeBlock.Snippet code={`hello javascript`} language="javascript" />
+        <CodeBlock.Snippet code={`hello php`} language="php" />
+      </CodeBlock>,
     );
-    expect(screen.getByText(/hello world/i)).toBeInTheDocument();
+    expect(screen.getByText("hello javascript")).toBeInTheDocument();
   });
 
-  it("should render line numbers", () => {
+  it("should support copy", async () => {
+    Object.assign(window.navigator.clipboard, {
+      writeText: vi.fn().mockImplementation(() => Promise.resolve()),
+    });
+
+    const { user } = render(
+      <CodeBlock language="javascript" onLanguageChange={() => {}}>
+        <CodeBlock.Header>Header</CodeBlock.Header>
+        <CodeBlock.Snippet code={`hello javascript`} language="javascript" />
+        <CodeBlock.Snippet code={`hello php`} language="php" />
+      </CodeBlock>,
+    );
+
+    const copyBtn = screen.getByRole("button", { name: /copy code/i });
+    await act(async () => await user.click(copyBtn));
+
+    expect(window.navigator.clipboard.writeText).toHaveBeenCalledWith(
+      "hello javascript",
+    );
+  });
+
+  it("should support language change", async () => {
+    const handleLanguageChange = vi.fn();
+
+    const { user } = render(
+      <CodeBlock language="php" onLanguageChange={handleLanguageChange}>
+        <CodeBlock.Header>Header</CodeBlock.Header>
+        <CodeBlock.Snippet code={`hello javascript`} language="javascript" />
+        <CodeBlock.Snippet code={`hello php`} language="php" />
+      </CodeBlock>,
+    );
+
+    expect(screen.getByText("hello php")).toBeInTheDocument();
+
+    const menuBtn = screen.getByRole("button", { name: /php/i });
+    await act(async () => await user.click(menuBtn));
+
+    const itemBtn = screen.getByRole("menuitem", { name: /node.js/i });
+    await act(async () => await user.click(itemBtn));
+
+    expect(handleLanguageChange).toBeCalledWith("javascript");
+  });
+
+  it("should support view only", async () => {
```

Comment:
Thought: My initial thought for what "view only" meant was that there was also no copy button, calling it something like "single language view" could make it easier to quickly understand